### DISCUSSION TITLE
Using ReplCache in production

### DISCUSSION THREAD

**Discussion by Huthesh** [belaban/JGroups#618] [General]
In my current project we are trying to use Jgroups for cluster. wanted to use a distributed cache. Initially we thought of using Jgroups Repl cache. But in this article http://www.jgroups.org/manual4/index.html#ReplCache its mentioned that "Note that this class was written as a demo of how state can be shared between nodes of a cluster. It has never been heavily tested and is therefore not meant to be used in production."
 
What is your opinion on using this class in production? What is the alternative to use?

**Comment by rhusar** [comment]
Use [Infinispan](https://infinispan.org/) instead.

**Reply by Huthesh** [reply]
In this link https://stackoverflow.com/questions/51844473/jgroups-replicatedhashmap-in-a-cluster @belaban mentioned we can use them in production. Could you please clarify?

**Reply by rhusar** [reply]
@Huthesh The sentence you pasted from documentation pretty much summarises the situation:

> "Note that this class was written as a demo of how state can be shared between nodes of a cluster. It has never been heavily tested and is therefore not meant to be used in production."

I suggested Infinispan since it's feature-rich, so once your requirements surpass that of ReplCache you can leverage a wide array of Infinispan's features. Furthermore, Infinispan has a commercial offering – https://www.redhat.com/en/technologies/jboss-middleware/data-grid – while there is no official commercial support available for `org.jgroups.blocks.ReplCache`.
